storage/queue.go

```diff
@@ -400,8 +401,12 @@ func (bq *baseQueue) remove(index int) {
 // Exposed for testing only.
 func (bq *baseQueue) DrainQueue(clock *hlc.Clock) {
 	bq.Lock()
-	defer bq.Unlock()
-	for repl := bq.pop(); repl != nil; repl = bq.pop() {
+	repl := bq.pop()
```

Comment:
If we're enqueuing things faster than we can process them then we're in trouble anyway, so I'm OK if tests time out in that case.